src/txmempool.h

```diff
@@ -736,6 +739,34 @@ class CTxMemPool
      *  removal.
      */
     void removeUnchecked(txiter entry, MemPoolRemovalReason reason) EXCLUSIVE_LOCKS_REQUIRED(cs);
+public:
+    // This function mutates mutable state!
```

Comment:
A bit confusing to have this comment (about a function) right before a class definition, I'm guessing it is leftover from something else (or should be moved a few lines down). 

Actually I think it'd make sense to explain here a little bit about what these epochs are and why/how to use them?  (Also, as this would be the first introduction of epochs in the code, it'd be helpful for reviewers of the future PRs that will use it in more places to be able to remember/understand what this construction is.)  Perhaps something like:

```
EpochGuard: RAII-style guard for using epoch-based graph traversal algorithms
When walking ancestors or descendants, we generally want to avoid visiting the 
same transactions twice. In some places we use std::set (or setEntries) to deduplicate
 what we visit, but we can do (algorithmically) better by using a counter ("epoch") that 
we set on each transaction and comparing against a global epoch to track whether 
we've visited something already during a calculation.
```
Or something like that...

Reply:
great comment; rewrote it slightly to a format I think is more clear & also to emphasize that replacement with epochs is ongoing work :)